### PR TITLE
Update cpe_vfuse

### PR DESCRIPTION
**Pull request by erikng**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This moves the template directory to a hash vs a list, so you can intelligently iterate through it and add manipulate settings in other API cookbooks.

This also configures the new vfuse 3.0 `settings.json` file so `vfuse --list-templates` will actually work with this cookbook.

Finally this brings in the `allow_downgrade` option, which by default forces idempotence, but an admin can override to `false` so they can install pre-release versions of vfuse without being downgraded.

**Which issue(s) this PR fixes**:
N/A

**Special notes for your reviewer**:
<3 @chilcote 

**Does this PR introduce a user-facing change?**:
Yes, it is a breaking change, as you will need to update your template code.

**Additional documentation e.g., Design Proposals, usage docs, etc.**:
N/A

### CODE REVIEW

**Comment by michel-slm** [comment]
thanks! we'll have to test internally and modify our local configuration; I'll update here if we have any concern.

File: itchef/cookbooks/cpe_vfuse/resources/darwin.rb [hunk 1]
```diff
@@ -74,4 +79,19 @@ def templates
...
     end
   end
+
+  def configure
+    template_dir = node['cpe_vfuse']['template_dir']
+    return if template_dir.nil?
+    config = {
+      'template_dir' => template_dir,
+    }
+    file '/usr/local/vfuse/bin/config.json' do
```
Comment (chilcote) [inline itchef/cookbooks/cpe_vfuse/resources/darwin.rb:89]: Can you add an only_if Dir.exist?('/usr/local/vfuse/bin') here? Not everyone is going to be installing vfuse into /usr/local/vfuse/bin.

**Comment by michel-slm** [comment]
I didn't notice this before, but ... @erikng can the config.json be in `#{base_dir}/etc/config.json` instead of in `bin` ? It's not an executable.

**Comment by michel-slm** [comment]
@chilcote just informed me that this clowniness of putting script and config in the same place is all thanks to upstream vfuse, and we'll have to fix it over there... so... I guess it's good for now. I'll rebase my config change on top of your updated PR and verify it works.

**Pull request closed without merging**